src/control/server/mgmt_client.go

```diff
@@ -59,6 +59,26 @@ func newMgmtSvcClient(ctx context.Context, log logging.Logger, cfg mgmtSvcClient
 	}
 }
 
+// delayRetry delays next retry.
+func (msc *mgmtSvcClient) delayRetry(ctx context.Context) {
+	delayCtx, cancelDelayCtx := context.WithTimeout(ctx, retryDelay)
+	select {
+	case <-delayCtx.Done():
+	}
+	cancelDelayCtx()
+}
+
+func (msc *mgmtSvcClient) logAttempt(prefix string, err error, status *int32) {
+	switch {
+	case err != nil:
+		msc.log.Debugf(prefix+": %v", err)
+	case *status != 0:
```

Comment:
To be safe, should check status != nil && *status != 0, I think.

Reply:
contemplated that but decided we would want to throw exception when something like this which shouldn't happen happens. will change